File: problems/1886-determine-whether-matrix-can-be-obtained-by-rotation/solution.cpp
class Solution {
public:
    bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
        int n = mat.size();
        for (int k = 0; k < 4; ++k) {
            if (mat == target) {
                return true;
            }
            vector<vector<int>> rotated(n, vector<int>(n));
            for (int i = 0; i < n; ++i) {
                for (int j = 0; j < n; ++j) {
                    rotated[j][n - 1 - i] = mat[i][j];
                }
            }
            mat = rotated;
        }
        return false;
    }
};

File: problems/1886-determine-whether-matrix-can-be-obtained-by-rotation/solution.py
class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        def rotate(matrix):
            n = len(matrix)
            new_matrix = [[0] * n for _ in range(n)]
            for i in range(n):
                for j in range(n):
                    new_matrix[j][n - 1 - i] = matrix[i][j]
            return new_matrix

        for _ in range(4):
            if mat == target:
                return True
            mat = rotate(mat)
        return False